### PR TITLE
feat(CRM): Opportunity Improvements and Vehicle Booking

### PR DESCRIPTION
**Pull request by haiderk0987**
> Please provide enough information so that others can review your pull request:

> Explain the **details** for making this change. What existing problem does the pull request solve?

> Screenshots/GIFs

### CODE REVIEW

File: erpnext/crm/doctype/opportunity/opportunity.js [hunk 1]
```diff
@@ -3,207 +3,275 @@
...
-				});
+		me.frm.set_query("opportunity_from", function() {
+			return {
+				"filters": {
+					"name": ["in", ["Customer", "Lead"]],
+				}
 			}
+		});
+
+		me.frm.set_query('party_name', function() {
```
Comment (haiderk0987) [inline erpnext/crm/doctype/opportunity/opportunity.js:101]: ouch...

File: erpnext/crm/doctype/lead/lead.py [hunk 2]
```diff
@@ -230,84 +195,60 @@ def set_lead_for_customer(lead, customer):
...
 	return target_doc
 
 
 @frappe.whitelist()
 def make_quotation(source_name, target_doc=None):
 	def set_missing_values(source, target):
-		_set_missing_values(source, target)
+		target.quotation_to = "Lead"
+		target.run_method("set_missing_values")
+		target.run_method("set_other_charges")
```
Comment (SaiFi0102) [inline erpnext/crm/doctype/lead/lead.py]: Need to decide between using `set_other_charges` or `append_taxes_from_master`

- set_taxes
- set_other_charges
- append_taxes_from_master

File: erpnext/crm/doctype/lead_existing_item/lead_existing_item.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
+{
+ "creation": "2022-10-24 17:21:08.197947",
+ "doctype": "DocType",
+ "editable_grid": 1,
```
Comment (SaiFi0102) [inline erpnext/crm/doctype/lead_existing_item/lead_existing_item.json:4]: Convert to child table

File: erpnext/crm/doctype/opportunity/opportunity.json [hunk 4]
```diff
@@ -9,67 +8,71 @@
  "editable_grid": 1,
  "engine": "InnoDB",
  "field_order": [
-  "from_section",
+  "title",
   "naming_series",
   "opportunity_from",
   "party_name",
   "customer_name",
```
Comment (SaiFi0102) [inline erpnext/crm/doctype/opportunity/opportunity.json:15]: Add display depends on for "customer_name"

File: erpnext/patches.txt [hunk 5]
```diff
@@ -780,3 +780,4 @@ execute:frappe.delete_doc_if_exists("Report", "Sales Payment Summary")
 execute:frappe.delete_doc_if_exists("Report", "Sales Order Items To Be Delivered")
 execute:frappe.delete_doc_if_exists("Report", "Purchase Order Items To Be Received")
 erpnext.patches.v12_0.set_booking_customer_name_in_jv
+erpnext.patches.v12_0.resetup_vehicle_domain #25-10-2022 #01-11-2022 #08-11-2022
```
Comment (SaiFi0102) [inline erpnext/patches.txt]: Keep it simple .. #25-10-2022 only

**Review by SaiFi0102 — CHANGES_REQUESTED** [review]

File: erpnext/patches.txt (hunk 5 above)
Comment (haiderk0987) [reply erpnext/patches.txt]: or... maybe.... 08-11-2022?

**Review by SaiFi0102 — APPROVED** [review]

**Comment by SaiFi0102** [comment]
ToDo:
- Add opportunity_item reference field in Vehicle Quotation and Vehicle Booking Order
- Check lead.py get_lead_details
- Set Customer reference in Lead instead of Lead reference in Customer
-

**Pull request merged**